src/composables/useImageUpload.ts

```diff
@@ -27,17 +27,28 @@ export function useImageUpload() {
     isUploadingImage.value = true;
     const formData = new FormData();
 
-    // TODO: Additional Validations - File Size, File Type, Empty File, Hidden File
     // TODO: Make this composable useFileUpload
 
     if (!['image/jpeg', 'image/jpg', 'image/png'].includes(file.type)) {
       imageUploadError.value = t('errors.unsupportedImageType');
       isUploadingImage.value = false;
       return;
     }
+    if (file.size > 1024 * 1024) {
+      imageUploadError.value = t('errors.fileTooBig');
+      isUploadingImage.value = false;
+      return;
+    }
     formData.append('file', file);
     try {
       const receipt = await pin(formData);
+
+      if (receipt.error) {
+        imageUploadError.value = receipt.error.message;
+        isUploadingImage.value = false;
+        return;
```

Comment:
Should we not throw an error instead of returning?

Reply:
I am keeping the same flow as the error from the file type